docker/android-system.sh

```diff
@@ -116,49 +559,264 @@ EOF
+    #   internal error: panic in GenerateBuildActions for module "com.android.example.apex" variant "android_common_com.android.example.apex_image"
+    # fixing this requires either a comprehensive removal of APEX from the build
+    # or adding numerous APEX dependencies, which defeats the purpose of a
+    # minimal bionic build.
+    sync art
+    sync bionic
+    sync build/blueprint
+    sync build/make
+    sync build/soong
+    sync external/apache-xml
+    sync external/bouncycastle
+    sync external/clang
+    sync external/compiler-rt
+    sync external/conscrypt
+    sync external/elfutils
+    sync external/fmtlib
+    sync external/golang-protobuf
+    sync external/gwp_asan
+    sync external/icu
+    sync external/jemalloc
+    sync external/jemalloc_new
+    sync external/libcxx
+    sync external/libcxxabi
+    sync external/libunwind
+    sync external/libunwind_llvm
+    sync external/llvm
+    sync external/lzma
+    sync external/mksh
+    sync external/okhttp
+    sync external/scudo
+    sync external/starlark-go
+    sync external/zlib
+    sync libcore
+    sync prebuilts/build-tools
+    sync prebuilts/clang-tools
+    sync prebuilts/clang/host/linux-x86
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.17-4.8
+    sync prebuilts/go/linux-x86
+    sync prebuilts/sdk
+    sync prebuilts/vndk/v28
+    sync prebuilts/vndk/v29
+    sync prebuilts/vndk/v30
+    sync system/core
+    sync system/libbase
+    sync system/linkerconfig
+    sync system/logging
+    sync system/sepolicy
+    sync system/tools/xsdc
+    sync tools/metalava
+    # these tools also seem to be required, since apex is now tightly
+    # coupled with the bionic build. unfortunately, we want to avoid
+    # building apex at all costs.
+    #sync system/apex
+    #sync system/tools/aidl
+
+    case "${ARCH}" in
+        arm)
+            sync external/arm-optimized-routines
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync external/arm-optimized-routines
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/tests/Android.mk
+    rm bionic/tests/Android.bp
+    rm bionic/benchmarks/Android.bp
+    rm bionic/tests/libs/Android.bp
+    rm bionic/tests/headers/Android.bp
+    rm bionic/tests/headers/posix/Android.bp
+
+    # make sure we don't build benchmarks or apex
+    rm -r bionic/apex
+    rm -r bionic/benchmarks/
+    rm -r bionic/tests/
+    rm -r system/linkerconfig/testmodules
+
+    remove_tests
+}
+
+remove_tests() {
+    install_packages python3-pip
+
+    local version=
+    version=$(python3 -c 'import sys
```

Comment:
The actual Python version may change with future Ubuntu updates, or if we upgrade to 22.04, etc. Since the Python path will differ accordingly for the installed packages, we need to detect the major and minor versions reliably. Luckily, this is very simple with `python -c $source_code`.